substrate/client/authority-discovery/src/worker.rs

```diff
@@ -168,6 +179,16 @@ pub struct Worker<Client, Block: BlockT, DhtEventStream> {
 	phantom: PhantomData<Block>,
 }
 
+#[derive(Debug, Clone)]
+struct RecordInfo {
+	// Time since UNIX_EPOCH in nanoseconds.
+	creation_time: u128,
+	// Peers that we know have this record.
+	peers_with_record: HashSet<PeerId>,
```

Comment:
Should we bound this to a decent number? We could have authority_keys * num_peers total?

Reply:
We shouldn't get more than 20(K_FACTOR) here, but you are right there is no hurt in explicitly bound it, so I'll do that.

Reply:
Bounded it to DEFAULT_KADEMLIA_REPLICATION_FACTOR, there is no need to over-complicate this with a LRU.